attribute/set.go

```diff
@@ -53,16 +58,44 @@ type (
 	Sortable []KeyValue
 )
 
-var (
-	// keyValueType is used in computeDistinctReflect.
-	keyValueType = reflect.TypeOf(KeyValue{})
+var slicePool = sync.Pool{New: func() any { return new([]KeyValue) }}
```

Comment:
I am curious how much this pool helps with performance. How much worse it is without this pool and with just allocating slices as needed?